lib/ansible/modules/network/nxos/nxos_file_copy.py

```diff
@@ -144,9 +143,9 @@ def transfer_file(module, dest):
     if not enough_space(module):
         module.fail_json(msg='Could not transfer file. Not enough space on device.')
 
-    hostname = module.params['host']
-    username = module.params['username']
-    password = module.params['password']
+    hostname = module.params['provider']['host']
```

Comment:
Not sure if this is a backward compatible change. 
How about making change as below?
```
provider = module.params['provider']
hostname = module.params['host'] or provider.get('host')
username = module.params['username'] or provider.get('username')
password = module.params['password'] or provider.get('password')
```